src/apps/replication/client_lib/replication_app_client_base.cpp

```diff
@@ -138,7 +151,7 @@ replication_app_client_base::request_context* replication_app_client_base::creat
     size_t offset = dsn_msg_body_size(request);
     ::marshall(request, rc->write_header);
 
-    if (rc->write_header.gpid.app_id == -1)
+    if (rc->write_header.gpid.pidx == -1)
```

Comment:
always true now or we will call get_partition_index(count, hash) if count is already known?